platform/darwin/src/MGLStyleLayer_Private.h

```diff
@@ -5,6 +5,25 @@
 
 #include <mbgl/style/layer.hpp>
 
+/**
+ Assert that the style layer is valid.
+
+ This macro should be used at the beginning of any public-facing instance method
+ of `MGLStyleLayer` and its subclasses. For private methods, an assertion is more appropriate.
+ */
+#define MGLAssertStyleLayerIsValid() \
+do { \
+if (!self.rawLayer) { \
+[NSException raise:@"Invalid style layer" \
+format: \
+@"-[%@ removeLayer:] has been called " \
```

Comment:
The method is named `-[MGLStyle removeLayer:]`, not `-[MGL*StyleLayer removeLayer:]`.